crates/network-tunnel/src/sshforwarding.rs

```diff
@@ -78,210 +72,134 @@ fn port_schema(title: &str, description: &str) -> schemars::schema::Schema {
 
 pub struct SshForwarding {
     config: SshForwardingConfig,
-    ssh_client: Option<Handle<ClientHandler>>,
-    local_listener: Option<TcpListener>,
+    process: Option<Child>,
 }
 
 impl SshForwarding {
-    const DEFAULT_SSH_PORT: u16 = 22;
-
     pub fn new(config: SshForwardingConfig) -> Self {
         Self {
             config,
-            ssh_client: None,
-            local_listener: None,
+            process: None,
         }
     }
 
-    pub async fn prepare_ssh_client(&mut self) -> Result<(), Error> {
-        let ssh_addrs =
-            Url::parse(&self.config.ssh_endpoint)?.socket_addrs(|| Some(Self::DEFAULT_SSH_PORT))?;
-        let ssh_addr = ssh_addrs.get(0).ok_or(Error::InvalidSshEndpoint)?;
-        let config = Arc::new(client::Config::default());
-        let handler = ClientHandler {};
-        self.ssh_client = Some(client::connect(config, ssh_addr, handler).await?);
-
-        Ok(())
-    }
-
-    pub async fn prepare_local_listener(&mut self) -> Result<(), Error> {
-        if self.config.local_port == 0 {
-            return Err(Error::ZeroLocalPort);
+    // We used to have `user` as a field on SSHForwarding config
+    // In order to be backward-compatible, we still allow that field, and if it exists we add
+    // the user ourselves manually
+    fn backward_compatible_forward_host(user: Option<&String>, forward_host: &String) -> String {
+        match user {
+            Some(user) => forward_host.replace("ssh://", &format!("ssh://{user}@")),
```

Comment:
This is not the right field to modify -- `forward_host` is the IP address of the target database, and `ssh_endpoint` is the one we want to add the user field to if present.

Reply:
I'm glad you noticed this, that was a late in the day commit 😁